File: simulation/bk_s3_2477_참외밭.java
package com.ssafy.im;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class bk_s3_2477_참외밭 {

	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int K = Integer.parseInt(br.readLine());

		int bean[] = new int[6];
		StringTokenizer st;
		int maxW = 0, maxH = 0;
		for (int i = 0; i < 6; i++) {
			st = new StringTokenizer(br.readLine());
			Integer.parseInt(st.nextToken());
			bean[i] = Integer.parseInt(st.nextToken());
			if (i % 2 == 0) {
				maxW = Math.max(bean[i], maxW);
			} else {
				maxH = Math.max(bean[i], maxH);
			}
		}
		int big = maxW * maxH;
		int smallW = 0, smallH = 0;
		for (int i = 0; i < 6; i++) {
			if (i % 2 == 0) {
				if (bean[(i + 5) % 6] + bean[(i + 1) % 6] == maxH) {
					smallW = bean[i];
				}
			} else {
				if (bean[(i + 5) % 6] + bean[(i + 1) % 6] == maxW) {
					smallH = bean[i];
				}
			}
		}
		int diff = smallW * smallH;
		System.out.println(K * (big - diff));
	}
}

File: 230412_PreviousData/simulation/bk_s3_2477_참외밭.py
K = int(input())
bean = []
maxW, maxH = 0, 0
for i in range(6):
    bean.append(int(input().split()[1]))
    if(i%2 ==0):
        maxW = max(bean[i], maxW)
    else:
        maxH = max(bean[i], maxH)
big = maxW * maxH
smallW, smallH = 0, 0
for i in range(6):
    if i % 2 == 0 :
        if bean[(i + 5) % 6] + bean[(i + 1) % 6] == maxH:
            smallW = bean[i]
    else:
        if bean[(i + 5) % 6] + bean[(i + 1) % 6] == maxW:
            smallH = bean[i]
diff = smallW * smallH
print(K * (big - diff))
